### PR TITLE
 Update NAT conntrack entries from natmgr instead of natorch

### PR DESCRIPTION
**Pull request by AkhileshSamineni**
Approach:

1. All NAT conntrack entries would be updated with large timeout (432000) value initially (when entry is created) by sending notifcation from NatSync to NatMgr.
2. All NAT conntrack enrries would be updated with Zero timeout value (when entry is deleted) be sending notification from OA to NatMgr.
3. Added a 1 day timer in NatMgr to reset the timeout for all conntrack entries to large time (43200) value.
4. NatOrch caches the time in secs whenever the entry is active, and compares it with specific timeout for every 30 secs when entry is not active.

Code changes:

NatSync:

1. Added a new timeout notification producer to send notifications to NatMgr.
2. Send a notification (whenever a new entry to created) to NatMgr to set 43200 time for that conntrack entry.

NatMgr:
1. While creating conntrack for Static entries, sets the timeout to max value (432000).
2. Added a new Refresh Timer for 1 day to set the timeout for all Static conntrack entries to max value (432000).
3. Added a new flush notification consumer for command “sonic-clear nat translations” and handling “Flushing all conntrack entries and added all existing Static conntrack entries” (This is moved from OA code).
4. Added an another new timeout notification consumer to handle timeout notifications from NatSync and NatOrch.
5. Setting the timeout for conntrack entries as per received notifications based on entry’s Key.

NatOrch:
1. Added a new variable for every entry, it caches the current time whenever entry becomes Active.
2. If entry is not Active, compares the cached time with current time and if it greater than global timeout, sends a age-out (0 timeout) notification to delete the conntrack entry.
3. Added a new timeout notification producer to send notification whenever an entry needs to be deleted.
4. Added a new Refresh Timer for 1 day to send SET (432000) timeout notifications to NatMgr for all dynamic entries.
 
Depends on:
https://github.com/Azure/sonic-utilities/pull/892
https://github.com/Azure/sonic-buildimage/pull/4596

Signed-off-by: Akhilesh Samineni <akhilesh.samineni@broadcom.com>

### CODE REVIEW

**Comment by lguohan** [comment]
need vs test for this feature.

**Comment by AkhileshSamineni** [comment]
> need vs test for this feature.
@lguohan  No new functionality added here, this PR is to address review comment ( https://github.com/Azure/sonic-swss/pull/1125#discussion_r347068029)

File: swssconfig/sample/00-copp.config.json [hunk 1]
```diff
@@ -55,5 +55,19 @@
             "red_action":"drop"
         },
         "OP": "SET"
+    },
```
Comment (arlakshm) [inline swssconfig/sample/00-copp.config.json:58]: Since this change is not related to the problem that is being solved in this PR. Can you move this change to a separate PR

**Comment by arlakshm** [comment]
> > need vs test for this feature.
> > @lguohan  No new functionality added here, this PR is to address review comment ( [#1125 (comment)](https://github.com/Azure/sonic-swss/pull/1125#discussion_r347068029))
@AkhileshSamineni I think we still need to add a unit test to the new logic to update the  conntrack timeouts

File: orchagent/natorch.cpp [hunk 2]
```diff
@@ -3269,6 +3269,70 @@ void NatOrch::queryHitBits(void)
...
+    while (natIter != m_natEntries.end())
+    {
+
+        if ((natIter->second.nat_type == "snat") and (natIter->second.addedToHw == true) and
+            (natIter->second.entry_type != "static"))
+        {
+            SWSS_LOG_ERROR("Update %s NAT entry [ip %s]", natIter->second.nat_type.c_str(), natIter->first.to_string().c_str());
+            std::vector<FieldValueTuple> fvVector;
+            std::string key = natIter->first.to_string();
+            setTimeoutNotifier->send("SET-SINGLE-NAT", key, fvVector);
```
Comment (arlakshm) [inline orchagent/natorch.cpp:3287]: It looks like the Timeout value is set to max for active entries in queryhitBits(), why do we need to update the timeout here again

File: swssconfig/sample/00-copp.config.json (hunk 1 above)
Comment (AkhileshSamineni) [reply swssconfig/sample/00-copp.config.json]: Yes, moved this change to new PR https://github.com/Azure/sonic-swss/pull/1300 .

**Comment by AkhileshSamineni** [comment]
> > > need vs test for this feature.
> > > @lguohan  No new functionality added here, this PR is to address review comment ( [#1125 (comment)](https://github.com/Azure/sonic-swss/pull/1125#discussion_r347068029))
> > > @AkhileshSamineni I think we still need to add a unit test to the new logic to update the  conntrack timeouts

@arlakshm Added a new test case to verify the conntrack entries timeout value.

File: orchagent/natorch.cpp (hunk 2 above)
Comment (AkhileshSamineni) [reply orchagent/natorch.cpp:3287]: The timeout is set to max value (432000) when entries got added in NatMgr code, in queryhitBits() we are deleting the entries which are not active. Here in this section of code (funtion updateAllConntrackEntries() ) called for every 86400 secs (1 day) to set all dynamic active entries conntrack timeout to Max value (432000).

**Comment by AkhileshSamineni** [comment]
Retest this please.

**Comment by abdosi** [comment]
@AkhileshSamineni Please create PR for 201911. Cherry-pick has conflict

@arlakshm @rlhui
